hub-templates/basehub/values.yaml

```diff
@@ -271,9 +271,43 @@ jupyterhub:
 
       06-custom-authenticator: |
         from oauthenticator.generic import GenericOAuthenticator
+        from jupyterhub.handlers import LogoutHandler
+        from tornado.httputil import url_concat
         from traitlets import Unicode
+        from z2jh import get_config
+
+        class CustomLogoutHandler(LogoutHandler):
+          """
+          Handle custom logout URLs. If a custom logout url
+          is specified, the 'logout' button will log the user out of that identity
+          provider in addition to clearing the session with Jupyterhub, otherwise
+          only the Jupyterhub session is cleared.
+          """
+
+          async def get(self):
+            await self.default_handle_logout()
+            await self.handle_logout()
+            if self.authenticator.logout_redirect_url:
```

Comment:
Let's do this conditional check inside render_logout_page instead? If all `super()` calls are only accesing the same function they are calling from, that makes life simpler I think.